pkg/bytecode/vm_test.go

```diff
@@ -28,12 +25,16 @@ func TestVMGlobals(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			tt.input += "\nx = x"
```

Comment:
I don't think you should be mutating the test cases. Instead create a local var and append to that. The risk is later the same test cases get used for some other tests and now the test case is wrong.

So just `input := tt.input + "\nx = x"` I think is fine.

This applies to all the tests changed in this way.